partner_invoicing_mode/models/res_partner.py

```diff
@@ -12,10 +12,21 @@ class ResPartner(models.Model):
         default=False,
         help="Do not group sale order into one invoice.",
     )
+    next_invoice_date = fields.Date(
+        help="This is the date at which the next invoice will be generated."
+    )
 
     @api.model
     def _commercial_fields(self):
         return super()._commercial_fields() + [
             "invoicing_mode",
             "one_invoice_per_order",
+            "next_invoice_date",
```

Comment:
This change requires to modify the `_update_next_invoice_date` method in the other modules to update only the commercial partner

Reply:
Why ?

Reply:
Because you update each partner of a company one by one and you trigger the sync on each contact of the company

Reply:
Yes; but that's the Odoo core behavior. You want to get wrong data values across partners of a same company ?

I

Reply:
Moreover, it is done one the whole recordset and not one by one :

https://github.com/odoo/odoo/blob/16.0/odoo/addons/base/models/res_partner.py#L619

Reply:
I was referring to this were you process each contact of a company one by one: https://github.com/OCA/account-invoicing/pull/1534#discussion_r1433953276